primitives/src/crh/poseidon/batched_crh.rs

```diff
@@ -240,296 +162,443 @@ impl<F: PrimeField + MulShort<F, Output = F>, P: PoseidonParameters<Fr = F>> Bat
 
-    #[test]
-    #[should_panic]
-    fn test_batch_hash_mnt6_null_elem() {
-        let input_batch = Vec::new();
-        let output_vec = (MNT6BatchPoseidonHash::batch_evaluate(&input_batch)).unwrap();
-        println!("{:?}", output_vec);
-    }
+            // =============================================================================
+            // Compare results
+            for i in 0..num_hashes {
+                assert_eq!(output_vec_in_place[i], output_vec[i], "Hash outputs, position {}, for MNT6 are not equal.", i);
+            }
+        }
 
-    #[test]
-    #[should_panic]
-    fn test_batch_hash_mnt6_one_elem() {
-        let mut input_batch = Vec::new();
-        input_batch.push(MNT6753Fr::from_str("1").unwrap());
-        let output_vec = (MNT6BatchPoseidonHash::batch_evaluate(&input_batch)).unwrap();
-        println!("{:?}", output_vec);
-    }
+        #[test]
+        #[should_panic]
+        fn test_batch_hash_mnt6_null_elem() {
+            let input_batch = Vec::new();
+            let output_vec = (MNT6BatchPoseidonHash::batch_evaluate(&input_batch)).unwrap();
+            println!("{:?}", output_vec);
+        }
 
-    #[test]
-    #[should_panic]
-    fn test_batch_hash_mnt6_three_elem() {
-        let mut input_batch = Vec::new();
-        input_batch.push(MNT6753Fr::from_str("1").unwrap());
-        input_batch.push(MNT6753Fr::from_str("2").unwrap());
-        input_batch.push(MNT6753Fr::from_str("3").unwrap());
-        let output_vec = (MNT6BatchPoseidonHash::batch_evaluate(&input_batch)).unwrap();
-        println!("{:?}", output_vec);
+        #[test]
+        #[should_panic]
+        fn test_batch_hash_mnt6_one_elem() {
+            let mut input_batch = Vec::new();
+            input_batch.push(MNT6753Fr::from_str("1").unwrap());
+            let output_vec = (MNT6BatchPoseidonHash::batch_evaluate(&input_batch)).unwrap();
+            println!("{:?}", output_vec);
+        }
+
+        #[test]
+        #[should_panic]
+        fn test_batch_hash_mnt6_three_elem() {
+            let mut input_batch = Vec::new();
+            input_batch.push(MNT6753Fr::from_str("1").unwrap());
+            input_batch.push(MNT6753Fr::from_str("2").unwrap());
+            input_batch.push(MNT6753Fr::from_str("3").unwrap());
+            let output_vec = (MNT6BatchPoseidonHash::batch_evaluate(&input_batch)).unwrap();
+            println!("{:?}", output_vec);
+        }
     }
 
-    #[test]
-    fn test_batch_hash_mnt4_in_place() {
-        //  the number of hashes to test
-        let num_hashes = 1000;
+    #[cfg(feature = "bn_382")]
+    mod bn_382 {
+        use super::*;
+        use algebra::fields::bn_382::{
+            Fr as BN382Fr,
+            Fq as BN382Fq,
+        };
+        use crate::{
+            BN382FrPoseidonHash, BN382FrBatchPoseidonHash,
+            BN382FqPoseidonHash, BN382FqBatchPoseidonHash,
+        };
+
+        #[test]
+        fn test_batch_hash_bn382fq() {
```

Comment:
What's the use of batch evaluation for the BN382 x^5-Poseidon?

Reply:
Ok, I have looked it up. For the BN382 we have the default implementation only, which just does a par.iter() over the batch.